sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1910,486 @@ def transform(self, X):
+
+    Parameters
+    ----------
+    n_quantiles : int, optional (default=1000)
+        Number of quantiles to be computed. It corresponds to the number
+        of landmarks used to discretize the cumulative density function.
+
+    output_distribution : str, optional (default='uniform')
+        Marginal distribution for the transformed data. The choices are
+        'uniform' (default) or 'norm'.
+
+    ignore_implicit_zeros : bool, optional (default=False)
+        Apply only for sparse matrices. If True, the sparse entries of the
+        matrix are discarded to compute the quantile statistics. If false,
+        these entries are accounting for zeros.
+
+    subsample : int, optional (default=1e5)
+        Maximum number of samples used to estimate the quantiles.
+
+    smoothing_noise : None or float, optional (default=None)
+        Standard deviation of the added noise before computing the
+        quantiles. This parameter is useful if there is a predominant
+        feature value predominant.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by np.random.
+
+    Attributes
+    ----------
+    quantiles_ : ndarray, shape (n_quantiles_, n_features)
+        The values corresponding the quantiles of reference.
+
+    Examples
+    --------
+    >>> from sklearn.datasets import load_iris
+    >>> from sklearn.preprocessing import QuantileTransformer
+    >>> iris = load_iris()
+    >>> X_trans = QuantileTransformer(n_quantiles=20).fit_transform(iris.data)
+
+    See also
+    --------
+    quantile_transform : Equivalent function without the object oriented API.
+
+    StandardScaler : perform standardization that is faster, but less robust
+    to outliers.
+
+    RobustScaler : perform robust standardization that removes the influence
+    of outliers but does not put outliers and inliers on the same scale.
+    """
+
+    def __init__(self, n_quantiles=1000, output_distribution='uniform',
+                 ignore_implicit_zeros=False, subsample=int(1e5),
+                 smoothing_noise=None, random_state=None):
+        self.n_quantiles = n_quantiles
+        self.output_distribution = output_distribution
+        self.ignore_implicit_zeros = ignore_implicit_zeros
+        self.subsample = subsample
+        self.smoothing_noise = smoothing_noise
+        self.random_state = random_state
+
+    def _compute_quantile_one_column(self, X_col, references, random_state):
+        """Private function to compute the quantiles for one features."""
+        if self.smoothing_noise is None:
+            noise = np.zeros(X_col.shape)
+        else:
+            noise = random_state.normal(0, self.smoothing_noise,
```

Comment:
why not just use `X_col = X_col +  ... ` here rather than allocating memory unnecessarily in the `smoothing_noise is None` case?